ACS-2809 Elasticsearch CMIS E2E tests for several comparators

dsibilio opened the pull request:
Adding E2E tests for `String`s/`Date`s/`Integer`s' comparators _(`=`, `<>`, `>`, `>=`, `<`, `<=`, `IN`, `NOT IN`)_.

Some of the comparators already work, others work assuming the fixes coming from [this enterprise-repo branch](https://github.com/Alfresco/alfresco-enterprise-repo/compare/feature/ACS-2809_support-comparators-against-different-field-types) are present. Below a summary of what already works + what works only with the aforementioned fixes.

|         | = | <> | > | >= | < | <= | IN | NOT IN |
|---------|---|----|---|----|---|----|----|--------|
| String  | W | W  | W | W  | W | W  | W  | W      |
| Date    | F | F  | W | W  | W | W  | F  | F      |
| Integer | F | F  | F | F  | F | F  | F  | F      |

**Legend:**
**W** = **W**orking natively
**F** = works, with the **F**ixes coming from https://github.com/Alfresco/alfresco-enterprise-repo/compare/feature/ACS-2809_support-comparators-against-different-field-types
**N** = **N**ot working

Commits up to and including [8418a55](https://github.com/Alfresco/acs-packaging/pull/2236/commits/8418a55fda605c547a29f0656752b573e1cf6cda) include all the tests for comparators that already work.
Commits after that include the rest of the tests.

KikoUK commented:
PR is still in Draft. Is it ready for reviews?

tpage-alfresco reviewed (approved):
This LGTM. I assume it's not draft any more?

dsibilio commented:
@tpage-alfresco @KikoUK 
> PR is still in Draft. Is it ready for reviews?
> This LGTM. I assume it's not draft any more?

Yes, it's ready for review. We figure we added all required tests. Updated the PR status.
